maskrcnn_benchmark/modeling/roi_heads/roi_heads.py

```diff
@@ -33,26 +34,27 @@ def forward(self, features, proposals, targets=None):
                 self.training
                 and self.cfg.MODEL.ROI_MASK_HEAD.SHARE_BOX_FEATURE_EXTRACTOR
             ):
-                mask_features = x
+                mask_features = box_features
             # During training, self.box() will return the unaltered proposals as "detections"
             # this makes the API consistent during training and testing
-            x, detections, loss_mask = self.mask(mask_features, detections, targets)
+            _, detections, loss_mask = self.mask(mask_features, detections, targets)
             losses.update(loss_mask)
 
         if self.cfg.MODEL.KEYPOINT_ON:
             keypoint_features = features
             # optimization: during training, if we share the feature extractor between
-            # the box and the mask heads, then we can reuse the features already computed
+            # the box and the keypoint heads, then we can reuse the features already computed
             if (
                 self.training
                 and self.cfg.MODEL.ROI_KEYPOINT_HEAD.SHARE_BOX_FEATURE_EXTRACTOR
             ):
-                keypoint_features = x
+                keypoint_features = box_features
             # During training, self.box() will return the unaltered proposals as "detections"
             # this makes the API consistent during training and testing
-            x, detections, loss_keypoint = self.keypoint(keypoint_features, detections, targets)
+            _, detections, loss_keypoint = self.keypoint(keypoint_features, detections, targets)
             losses.update(loss_keypoint)
-        return x, detections, losses
+
+        return detections, losses
```

Comment:
even if this is not used in the rest of the code, this is something that should be discussed further, and is a breaking change.

Can you revert this?

Reply:
Hi @fmassa, you are right. I've reverted these two changes you mentioned. `box_features` from box head will be returned. I'm also happy to have further discussion with you regarding on this change.